common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -95,6 +111,38 @@ public static String getRemoteAddress(Channel channel) {
     return "<unknown remote>";
   }
 
+  /**
+   * Returns the default number of threads for both the Netty client and server thread pools.
+   * If numUsableCores is 0, we will use Runtime get an approximate number of available cores.
+   */
+  public static int defaultNumThreads(int numUsableCores) {
+    final int availableCores;
+    if (numUsableCores > 0) {
+      availableCores = numUsableCores;
+    } else {
+      availableCores = Runtime.getRuntime().availableProcessors();
+    }
+    return Math.min(availableCores, MAX_DEFAULT_NETTY_THREADS);
+  }
+
+  /**
+   * Returns the lazily created shared pooled ByteBuf allocator for the specified allowCache
+   * parameter value.
+   */
+  public static synchronized PooledByteBufAllocator getSharedPooledByteBufAllocator(
+      boolean allowDirectBufs,
```

Comment:
Not sure it's a big deal, but after the allocator is cached, it won't matter what this argument is; it will be ignored.

Reply:
That's not quite the issue here, it's that the cached instance that this returns may not reflect the value that was passed to the method. I don't know from the callers whether that's a real issue or not. By itself this doesn't require any new configs.

Reply:
I still think a non-module based config would solve this issue.

Otherwise we could end up 4 PooledByteBufAllocator's. Like
- two (client and server) for rpc where `preferDirectBufs=false`
- two for shuffle  where `preferDirectBufs=true`